doc.config

```diff
@@ -0,0 +1,18 @@
+%% ./bootstrap
+%% REBAR_CONFIG=doc.config ./rebar3 ex_doc
```

Comment:
This works and I'm not opposed to it. I think this method omits the _rest_ of the Rebar3 config when working on the project, which may drop some settings. Was this on purpose because they might clash?

If it wasn't on purpose, then one option here would have been to use profiles for this whole section within the default `rebar.config` file, and then allow calling `./rebar3 as docs ex_doc`. This would require unscripting the stuff in `doc.config.script` though, which would be way too much work for a demo I imagine, but might be worth it in the end if we go for full support.

Reply:
I did it with doc.config[.script] on purpose for the first draft to demonstrate the blast radius of the change. Specifically, that the ex_doc config should be expected to have no impact on the rest of the project, to alleviate concerns about circular dependencies. I think we're on the same page here, so this is no longer a goal for the PR. 👍 

I can continue the effort by exploring what it looks like in a merged rebar.config and without the script. However, without the script I think this will mean listing all markdown files explicitly in the rebar.config. I'll make it so and we can see how unmanageable that list looks. I don't mind doing multiple iterations to get it right.